feat: integrate with @crossbell/ipfs-gateway

runjuu opened the pull request:
### [@crossbell/ipfs-gateway-next](https://github.com/Crossbell-Box/crossbell-monorepo/blob/main/packages/ipfs-gateway-next/src/index.ts)
A Next.js plugin. It is used to serve a static file that service worker needs. It also provides a fallback strategy to handle `/_ipfs/{cid}` request which just redirects the request to other valid gateways.

### [@crossbell/ipfs-gateway](https://github.com/Crossbell-Box/crossbell-monorepo/tree/main/packages/ipfs-gateway/src)
Including service worker registration and some helper methods.

### [@crossbell/ipfs-react](https://github.com/Crossbell-Box/crossbell-monorepo/tree/main/packages/ipfs-react/src)
Provide first class `ipfs-gateway` support for react ecosystem 🤡.

runjuu committed:
chore: ignore IDE specific files

runjuu committed:
feat: integrate with `@crossbell/ipfs-gateway`

DIYgod committed:
fix: sw.js not found on subdomain

DIYgod commented:
Is this feature still under development or has been completed?

Runjuu commented:
The feature has been completed, but related README has not 😅. It should be ok to merge, I think.

DIYgod commented:
I found two problems
1. the server cannot access the `_ipfs` path, one option is to have the server access `SITE_URL + "/_ipfs/"` and the other is to have the server access the normal gateway
2. it always falls back to `cf-ipfs` under the subdomain, you can locally run `pnpm dev` and then visit `http://diygodbest.localhost:3000/` to see

runjuu committed:
fix(ipfs-gateway): server side request issues

- `Unidata` will use default gateway when running at server side.
- `toGateway` adds an option `needRequestAtServerSide` to return a valid gateway for server side usage.
